src/osd/ECBackend.cc

```diff
@@ -1686,6 +1686,29 @@ int ECBackend::get_min_avail_to_read_shards(
   return 0;
 }
 
+void ECBackend::get_min_want_to_read_shards(
+  pair<uint64_t, uint64_t> off_len,
+  set<int> *want_to_read) {
+  uint64_t off = off_len.first;
+  uint64_t len = off_len.second;
+  uint64_t chunk_size = sinfo.get_chunk_size();
+  int data_chunk_count = sinfo.get_stripe_width() / sinfo.get_chunk_size();
+  const vector<int> &chunk_mapping = ec_impl->get_chunk_mapping();
+
+  int total_chunks = (chunk_size - 1 + len) / chunk_size;
+  int first_chunk = (off / chunk_size) % data_chunk_count;
+
+  if (total_chunks > data_chunk_count) {
+    total_chunks = data_chunk_count;
+  }
+
+  for(int i = 0; i < total_chunks; i++) {
+    int j = (first_chunk + i) % data_chunk_count;
```

Comment:
please add a comment to explain the logic here (and maybe rename 'j'?)

Reply:
left the "j" for now. We have a similar loop in get_want_to_read_shards here:
https://github.com/ceph/ceph/blob/main/src/osd/ECBackend.h#L211-L217

If we are going to change these around I think we should perhaps do it at the same time for both functions in another PR.  I did add documentation for this function in the header though.

Reply:
This is among the less readable files in the OSD code base, let's not double down on my prior poor taste.  i, j, and k are fine as simple indices for short for loops, but j isn't a simple index.  It should probably be chunk_to_read.